packages/app/src/app/pages/common/Modals/SubscriptionCancellation/SubscriptionCancellation.tsx

```diff
@@ -0,0 +1,160 @@
+  line-height: 1;
+  color: #ef7a7a;
+`;
+
+export const SubscriptionCancellationModal: React.FC = () => {
+  const { activeTeamInfo } = useAppState();
+  const { isTeamAdmin } = useWorkspaceAuthorization();
+  const { hasActiveTeamTrial } = useWorkspaceSubscription();
+  const { modalClosed } = useActions();
+  const [loadingCustomerPortal, createCustomerPortal] = useCreateCustomerPortal(
+    {
+      team_id: isTeamAdmin ? activeTeamInfo?.id : undefined,
+    }
+  );
+  // TO DO: if we need this info more often, extract
+  // to a custom useWorkspaceUsage similar to
+  // useWorkspaceLimits.
+  const teamUsage = activeTeamInfo?.usage;
+
+  const getEditorsLabel = () => {
+    if (teamUsage && teamUsage.editorsQuantity > TEAM_FREE_LIMITS.editors) {
+      const lostSeats = teamUsage.editorsQuantity - TEAM_FREE_LIMITS.editors;
+      return `-${lostSeats} editor seat${lostSeats === 1 ? '' : 's'}`;
+    }
+
+    return `Limited to ${TEAM_FREE_LIMITS.editors} editors seats`;
+  };
+
+  // Private sandboxes are a Pro feature only. If the
+  // team downgrades, these will become restricted.
+  // Public sandboxes do not get restricted on
+  // downgrading but are limited to a certain amount
+  // if the team does not have an active subscription.
+  const getSandboxesLabel = () => {
+    const restrictedSandboxesCount =
+      (teamUsage?.privateProjectsQuantity ?? 0) -
+      TEAM_FREE_LIMITS.private_repos;
+    if (restrictedSandboxesCount > 0) {
+      return `${restrictedSandboxesCount} sandbox${
+        restrictedSandboxesCount === 1 ? '' : 'es'
+      } will be restricted`;
+    }
+
+    return `Limited to ${TEAM_FREE_LIMITS.public_sandboxes} public sandboxes`;
+  };
+
+  // Private repositories are a Pro feature only. If the
+  // team downgrades, these will become restricted.
+  // Public repositories do not get restricted on
+  // downgrading but are limited to a certain amount
+  // if the team does not have an active subscription.
+  const getRepositoriesLabel = () => {
+    const restrictedRepositoriesCount =
+      (teamUsage?.privateProjectsQuantity ?? 0) -
+      TEAM_FREE_LIMITS.private_repos;
+    if (restrictedRepositoriesCount > 0) {
+      return `${restrictedRepositoriesCount} repositor${
+        restrictedRepositoriesCount === 1 ? 'y' : 'ies'
+      } will be restricted`;
+    }
+
+    return `Limited to ${TEAM_FREE_LIMITS.public_repos} public repositories`;
+  };
+
+  return (
+    <Stack
+      css={{
+        position: 'relative',
+        overflow: 'hidden',
+        padding: '24px 32px 32px',
+      }}
+      direction="vertical"
+    >
+      <Stack>
+        <Text
+          as="h1"
+          css={{
+            margin: 0,
+            lineHeight: '28px',
+            letterSpacing: '-0.019em',
+          }}
+          size={19}
+          weight="normal"
+        >
+          You&apos;ll lose access to all Pro features if you decide to cancel
+        </Text>
+        <IconButton name="cross" title="Close" />
```

Comment:
Button is not connected to the close operation

Reply:
Addressed here: [9b9f682](https://github.com/codesandbox/codesandbox-client/pull/7195/commits/9b9f6823b734a1424d04e7e586562c66c78f80ef).